gcp_variant_transforms/vcf_to_bq.py

```diff
@@ -218,44 +219,73 @@ def run(argv=None):
+        known_args.partition_config_path)
+
   beam_pipeline_options = pipeline_options.PipelineOptions(pipeline_args)
   pipeline = beam.Pipeline(options=beam_pipeline_options)
   variants = _read_variants(pipeline, known_args)
   variants |= 'FilterVariants' >> filter_variants.FilterVariants(
       reference_names=known_args.reference_names)
+  if partitioner:
+    num_partitions = partitioner.get_num_partitions()
+    variants = variants | 'PartitionVariants' >> beam.Partition(
+        partition_variants.PartitionVariants(partitioner), num_partitions)
+    if (not partitioner.should_flatten() and
+        not partitioner.has_residual_partition()):
+      #  An extra partition for residual variants is added to the end of list.
+      #  This extra partition will be ignored in the rest of the pipeline.
+      num_partitions -= 1
+      variants = [variants[i] for i in range(0, num_partitions)]
+  else:
+    # By default we don't partition data, so we have only 1 partition.
+    num_partitions = 1
+    variants = [variants]
+
   if variant_merger:
-    if known_args.optimize_for_large_inputs:
-      partitions = variants | 'PartitionVariants' >> beam.Partition(
-          partition_variants.PartitionVariants(partitioner),
-          partitioner.get_num_partitions())
-      merged = []
-      for i in xrange(partitioner.get_num_partitions()):
-        merged.append(partitions[i] | 'MergeVariants' + str(i) >>
-                      merge_variants.MergeVariants(variant_merger))
-      variants = merged | 'FlattenPartitions' >> beam.Flatten()
-    else:
-      variants |= ('MergeVariants' >> merge_variants.MergeVariants(
-          variant_merger))
-  proc_variants = variants | 'ProcessVaraints' >> beam.Map(
-      processed_variant_factory.create_processed_variant).\
-    with_output_types(processed_variant.ProcessedVariant)
-  _ = (proc_variants |
-       'VariantToBigQuery' >> variant_to_bigquery.VariantToBigQuery(
-           known_args.output_table,
-           header_fields,
-           variant_merger,
-           processed_variant_factory,
-           append=known_args.append,
-           allow_incompatible_records=known_args.allow_incompatible_records,
-           omit_empty_sample_calls=known_args.omit_empty_sample_calls,
-           num_bigquery_write_shards=known_args.num_bigquery_write_shards))
+    variants = apply_transform(
+        variants, num_partitions, 'MergeVariants',
+        merge_variants.MergeVariants(variant_merger))
+    if partitioner.should_flatten():
+      variants = [variants | 'FlattenPartitions' >> beam.Flatten()]
+      num_partitions = 1
+  proc_variants = apply_transform(
+      variants, num_partitions, 'ProcessVaraints',
+      beam.Map(
+          processed_variant_factory.create_processed_variant).with_output_types(
+              processed_variant.ProcessedVariant))
+  for i in range(num_partitions):
+    partition_name = partitioner.get_partition_name(i)
```

Comment:
Also need to take into account when partitioner is None. With the previous suggestion about moving the `_` to here, I suggest something like:
```
table_suffix = '_' + partitioner.get_partition_name(i) if partitioner else ''
```

Reply:
done (except that I used a normal, easier to read if statement).